compliance_checker/cf/cf.py

```diff
@@ -2952,34 +2954,33 @@ def check_climatological_statistics(self, ds):
                         boundary_variable.dimensions)
                 )
 
-        # otherwise match the following values with for variable with
-        # `cell_methods` attributes
-        # time: method1 within years time: method2 over years
-        # time: method1 within days time: method2 over days
-        # time: method1 within days time: method2 over days time: method3 over years
-        # optionally followed by parentheses for explaining additional
-        # info, e.g.
-        # "time: method1 within years time: method2 over years (sidereal years)"
-
-        meth_regex = "(?:{})".format("|".join(methods))
-        re_string = (r'^time: {0} within (years|days)'
-                     r' time: {0} over \1(?<=days)(?: time: {0} over years)?'
-                     r'(?: \([^)]+\))?$'.format(meth_regex))
-
-        # find any variables with a valid climatological cell_methods
-        for cell_method_var in ds.get_variables_by_attributes(cell_methods=lambda s: s is not
-                                                              None):
-            total_climate_count += 1
-            if not regex.search(re_string, cell_method_var.cell_methods):
-                reasoning.append('The "time: method within years/days over years/days" format is not correct in variable {}.'.format(cell_method_var.name))
-            else:
-                valid_climate_count += 1
+            # match the following values with for variable with
+            # `cell_methods` attributes
+            # time: method1 within years time: method2 over years
+            # time: method1 within days time: method2 over days
+            # time: method1 within days time: method2 over days time: method3 over years
+            # optionally followed by parentheses for explaining additional
+            # info, e.g.
+            # "time: method1 within years time: method2 over years (sidereal years)"
+
+            meth_regex = "(?:{})".format("|".join(methods))
+            re_string = (r'^time: {0} within (years|days)'
+                         r' time: {0} over \1(?<=days)(?: time: {0} over years)?'
+                         r'(?: \([^)]+\))?$'.format(meth_regex))
+
+            # find any variables with a valid climatological cell_methods
+            for cell_method_var in ds.get_variables_by_attributes(cell_methods=lambda s: s is not None):
```

Comment:
Strictly speaking this check should only be applied to variables that have `clim_var`  as a coordinate. (CF doesn's say you can't have non-climatological variables in the same file -- though it would probably be unwise to do so.)